src/exchangeRateRouter.ts

```diff
@@ -25,7 +32,21 @@ const asExchangeRatesReq = asObject({
   data: asArray(asExchangeRateReq)
 })
 
+const asExtendedReq = <T>(asExtension: CleanerShape<T>) => (
+  raw: any
+): express.Request & T => {
+  const extension = asObject(asExtension)(raw)
+  Object.assign(raw, extension)
+  return raw
+}
+
+const asExtendedRequest = asExtendedReq({
+  requestedRates: asMaybe(asExchangeRatesReq),
+  requestedRatesResult: asMaybe(asReturnGetRate)
+})
```

Comment:
Short fixup:

```ts
const asRatesRequest = asExtendedReq({
  requestedRates: asOptional(asExchangeRatesReq),
  requestedRatesResult: asOptional(asReturnGetRate)
})
```

1. Name reflects the rates endpoints.
2. Use `asOptional` over `asMaybe` because we only want to allow `undefined` or the cleaner type; anything other would throw.